### PR TITLE
Refactored to save settings only if value changes

### PR DESCRIPTION
**Pull request by ErisDS**
This is quite a small change, but with potentially larger consequences. At the moment, requests from the Ghost Admin come into the settings endpoint with multiple settings to update. There's no guarantee any of them have changed.

On the server side, we blindly processed all the settings, saving each one with the "new" value regardless of whether anything changed. The updated_at value will change every time.

The code in this PR changes this to be smarter. We set relevant properties on the model and then if bookshelf's `hasChanged` method tells us any of the values are different, we save, else we just return.

The benefit is less events flying through the settingsCache system. My end goal here related to #9192 is to know when a setting is _really_ changed to check if a route like `/subscribers/`or `/amp/` should be enabled or not.

refs #9192

- Each setting is saved individually
- Update this to only happen on import, or when a value changes
- Reduces the amount of work Ghost does on every setting change

### CODE REVIEW

**Comment by ErisDS** [comment]
@kirrg001 & @kevinansfield I have tested this and it seems to work well. Just wanted to quickly check - can either of you think of anything this might break? That's all I want review for 😄

**Comment by kevinansfield** [comment]
> can either of you think of anything this might break?

Nothing that I can think of 🤔 Sounds like a sane change to me!

**Review by kevinansfield — APPROVED** [review]

File: core/server/models/settings.js [hunk 1]
```diff
@@ -118,22 +118,25 @@ Settings = ghostBookshelf.Model.extend({
...
+                        if (item.hasOwnProperty('value')) {
+                            setting.set('value', item.value);
+                        }
+                        // Internal context can overwrite type (for fixture migrations)
+                        if (options.context && options.context.internal && item.hasOwnProperty('type')) {
+                            setting.set('type', item.type);
+                        }
+
+                        if (setting.hasChanged()) {
+                            return setting.save(options);
```
Comment (kirrg001) [inline core/server/models/settings.js]: This must be `save(null, options)`.

**Review by kirrg001 — CHANGES_REQUESTED** [review]
Using `settings.set` allows us to use `hasChanged`, makes perfectly sense 👍  Good improvement.
I just added one comment.

**Comment by ErisDS** [comment]
@kirrg001 good catch, pushed a fix

**Review by kirrg001 — APPROVED** [review]
👍  Feel free to self merge if travis is green!

**Pull request merged**